component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateUtil.java

```diff
@@ -114,8 +124,6 @@ public static boolean addCertificate(String username, byte[] certificateBytes)
         } catch (CertificateException e) {
             throw new AuthenticationFailedException("Error while retrieving certificate of user: " + username, e);
         } catch (UserStoreException e) {
-            throw new AuthenticationFailedException("Error while setting certificate of user: " + username, e);
-        } catch (org.wso2.carbon.user.api.UserStoreException e) {
             throw new AuthenticationFailedException("Error while user manager for tenant id", e);
```

Comment:
"Error while user manager for tenant id" is not meaningful . Hence change the error message as "Error while retrieving the user store manager for tenant id  " or as something meaningful.